### PR TITLE
Add lat lon validation

### LINKED ISSUE #164: Invalid longitude/latitude error message
#157
We need to check the latitude/longitude the user inputs to ensure they are valid, and if not, display an error message and prevent the user from inserting/updating a device.

### PR DESCRIPTION
**Pull request by thessbsu**
Closes #164 

If an invalid entry is present it will be flagged when clicking "Add Device"
image

### CODE REVIEW

**Comment by thessbsu** [comment]
fixed "merge into" branch

File: BEIMA.Client/src/pages/Devices/AddDevicePage.js [hunk 1]
```diff
@@ -19,28 +20,38 @@ const AddDevicePage = () => {
...
     let fieldValues = {};
+    let newErrors = {};
 
     for(let i = 0; i < formFields.length; i++){
       let formName = formFields[i].name;
       let fieldNames = Object.keys(deviceFields);
 
       if(fieldNames.includes(formName)){
         let formJSON =  {[formName] : formFields[i].value};
-        formFields[i].value = "";
```
Comment (AshlynAdamson) [inline BEIMA.Client/src/pages/Devices/AddDevicePage.js:43]: Can we keep this line, but move it to a better spot? I use it to clear the fields on a submit. Or maybe we can create a separate method that clears the fields after all the input validation is done and passes

Comment (thessbsu) [reply BEIMA.Client/src/pages/Devices/AddDevicePage.js:43]: I had tried to post a reply earlier but it doesn't look like it took.

Clearing the from will be in the Else statement when detecting errors. I'm hoping that setting the DB JSON object for the fields by selected device type to the currentDeviceFields variable will clear it. I've got some in progress stuff for that in another (unpublished at the moment) branch

**Review by AshlynAdamson — APPROVED** [review]
Looks good! Glad you got the formlist to work

**Review by josephmoore99 — APPROVED** [review]

**Review by roguedeleter — APPROVED** [review]

**Pull request merged**